### PR TITLE
uplift job income page to bs4

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [x] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [x] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [x] Code does not use .html_safe
- [x] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/n/projects/2640062/stories/187140851
Ticket: https://www.pivotaltracker.com/n/projects/2640062/stories/187717414

# A brief description of the changes

Current behavior: No uplift to bs4 for FAA job incomes page, outdated setup of view partials, clean up of translations

New behavior: Uplifted, improved end date warnings, updated partials and translations

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

**There may be some more changes to this once Charlie's [PR](https://github.com/ideacrew/enroll/pull/3954) has been merged**

Screenshot 2024-06-25 at 2 45 25 PM
Screenshot 2024-06-25 at 2 45 40 PM
Screenshot 2024-06-25 at 2 45 54 PM
Screenshot 2024-06-25 at 3 16 38 PM
Screenshot 2024-06-25 at 3 16 43 PM

### CODE REVIEW

**Review by scaustin34 — COMMENTED** [review]
it looks like the radio buttons are still interact-able when they're greyed out, causing you to be able to make multiple instance of the new_income form that continue to stack infinitely, we should probably truely disable them when they're grey

File: app/javascript/css/forms.scss [hunk 1]
```diff
@@ -217,6 +217,14 @@ input[type='date']:invalid::-webkit-datetime-edit {
   overflow: hidden;
 }
 
+.income-padding {
```
Comment (charlienparker) [inline app/javascript/css/forms.scss:220]: I was able to get the forms/fake forms to match up in width without adding these classes. Do you need these classes?

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 2]
```diff
@@ -1,33 +1,42 @@
...
-  $('a.new-income').addClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').attr('disabled', true);
+  $('#self-employed-income').addClass('disabled');
+  $('a.new-income').addClass('disabled');
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').attr('disabled', true);
   $("a.interaction-click-control-add-more").addClass('hide'); // legacy
 
   // disable all income edit and edit buttons on created incomes
   $("a[class*='income-edit'], a[class*='income-delete']").addClass('disabled');
+  $('a.income-edit').addClass('disabled');
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:37]: the above selector `"a[class*='income-edit'], a[class*='income-delete']` should already match these two you added. do you need these two?

File: components/financial_assistance/app/assets/stylesheets/financial_assistance/financial_assistance.scss [hunk 3]
```diff
@@ -383,7 +383,9 @@ div#add-more-link {
 .mt-label{
   margin-top: -19px;
 }
-
+.mt-auto {
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/stylesheets/financial_assistance/financial_assistance.scss:386]: is this new class needed? I didn't need this class for my radio buttons

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb [hunk 4]
```diff
@@ -137,11 +136,15 @@ def enable_bs4_layout
       @bs4 = true
     end
 
+    def conditionally_enable_bs4_layout
```
Comment (charlienparker) [inline components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:139]: you can remove this method, it's already defined above

File: components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb [hunk 5]
```diff
@@ -1,212 +1,431 @@
...
-        </div>
-        <div class='col-md-6 left-seprator'>
-          <a href="#has_job_income" data-toggle="modal" data-target="#has_job_income">Not sure?</a>
-          <%= render partial: 'financial_assistance/shared/modal_support_text', locals: {key: "has_job_income"} %>
-        </div>
+      <fieldset>
+        <legend id="job_income_value" class='income-padding focus'>
+          <div class='d-inline-flex'>
+            <%= radio_button_tag("has_job_income", true, @applicant.has_job_income, class: "radio-yml mt-auto" ) %>
+            <label class="radio-yml" for="has_job_income_true"><%= l10n("yes") %></label>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb:18]: add weight-n to this label and below for the other radio

File: components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb [hunk 6]
```diff
@@ -1,212 +1,431 @@
...
       </div>
-    </div> <br>
+    </div>
 
-    <div class="driver-question row row-form-wrapper radio-align no-buffer fa-align-elements">
-      <div class='col-md-6 labelforcoverage'>
-        <span class="fa-text-color"><%= l10n('faa.incomes.from_self_employment', assistance_year: assistance_year) %>*</span>
+    <div class="mb-3 mt-5 income-padding driver-question">
+      <div class='labelforcoverage'>
+        <span class="fa-text-color required"><%= l10n('faa.incomes.income_from_self', subject: @applicant.first_name) %></span>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb:47]: missing key

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb [hunk 7]
```diff
@@ -1,78 +1,145 @@
...
+    <%= hidden_field_tag :bs4, true %>
+    <% income_id = income.id %>
+    <% onchange_date = FinancialAssistanceRegistry[:income_and_deduction_date_warning].enabled? ? (raw "validateDateWarnings('#{income_id}', true)") : "" %>
+    <%= f.hidden_field :kind, value: 'wages_and_salaries' %>
+    <div class="alert alert-info d-flex" role="alert">
+      <div class="pl-2 pt-2 pr-1">
+        <div class="info-icon icon" alt="info" aria-hidden="true">&nbsp;</div>
+      </div>
+      <div class="mt-1 pl-0 pr-2 pt-1">
+        <p class="subtext"><%= l10n('faa.incomes.job_end_date_note', subject: @applicant.first_name, income_type: "job") %></p>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb:12]: Can we reuse the `faa.incomes.income_note` for this content? If so, we can get rid of this key in the translations file as well

File: components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income_form.erb [hunk 8]
```diff
@@ -1,40 +1,87 @@
...
+<% if @bs4 %>
+  <%= income_form_for @application, @applicant, income do |f| %>
+    <%= hidden_field_tag :bs4, true %>
+    <div class="my-4">
+      <div class="alert alert-info d-flex" role="alert">
+        <div class="pl-2 pt-2 pr-1">
+          <div class="info-icon icon" alt="info" aria-hidden="true">&nbsp;</div>
+        </div>
+        <div class="mt-1 pl-0 pr-2 pt-1">
+          <p class="subtext"><%= l10n('faa.incomes.job_end_date_note', subject: @applicant.first_name, income_type: "self-employment") %></p>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income_form.erb:10]: Can we reuse the faa.incomes.income_note for this content? If so, we can get rid of this key in the translations file as well

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:1]: I think you should be calling the new method `generateNewForm` whenever the app renders a new form, ie, whenever the user checks the inputs for self-employed or job income OR when they Add Additional Income, otherwise there will be WAVE errrors

**Review by charlienparker — CHANGES_REQUESTED** [review]
Looks good, mostly small tweaks - but still need to fix the WAVE errors following the dummy approach from my PR

File: components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb [hunk 9]
```diff
@@ -1,212 +1,431 @@
...
     </div>
 
     <div id='income-error-popup'></div>
 
-    <!-- Hidden unsaved income modal -->
-    <div class="modal fade" id="unsavedIncomeChangesWarning" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+    <%= render partial: 'financial_assistance/shared/end_date_warning' %>
+
+    <!-- Hidden unsaved benefits modal -->
+    <div class="modal fade" id="unsavedBenefitChangesWarning" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb:85]: this modal does not appear to be showing when there are incomes being edited/created

File: components/financial_assistance/app/views/financial_assistance/shared/_modal_support_text.html.erb
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/shared/_modal_support_text.html.erb:1]: this file has an error on my IDE

File: components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb [hunk 10]
```diff
@@ -287,6 +289,19 @@
...
+  "en.faa.incomes.remove_all" => 'Are you sure you want to remove all job income(s)?',
+  "en.faa.incomes.confirm_deletion" => 'Confirm Deletion',
+  "en.faa.incomes.job_income_message" => "Tell us about job income for %{subject}, select “Continue to Next Step” when finished.",
+  "en.faa.incomes.job_end_date_note" => "Note, for %{income_type} income %{subject} currently receives, do not enter an end date into the “End Date” field, only enter an end date if the %{income_type} income has ended.",
+  "en.faa.incomes.from_employer_question" => "Does %{subject} have income from an employer?",
+  "en.faa.incomes.end_date_warning" => "Leave the “%{end_date_field_label}” field blank unless your income will end on a specific date (e.g. the date your seasonal job ends). Only enter the last day of the current year if you won't have this income next year.",
+  "en.faa.incomes.end_date_header" => "End Date:",
+  "en.faa.end_date_warning" => "<strong>End Date:</strong> Leave the 'To' field blank unless your income will end on a specific date (e.g. the date your seasonal job ends). Only enter the last day of the current year if you won't have this income next year.",
```
Comment (charlienparker) [inline components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb:304]: is this string needed? We already have the `faa.incomes.end_date_warning` translation which should work for our reqs

File: components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb:1]: is this file needed?

File: components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        </div>
+        <div class="col-auto px-0">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <div class="close-icon icon align-self-start m-0" aria-hidden="true">&nbsp;</div>
+          </button>
+        </div>
+      </div>
+      <div class="modal-body">
+        <b><%= l10n("faa.incomes.end_date_header") %></b>
+        <%= l10n("faa.incomes.end_date_reminder")  %>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb:19]: this key doesn't exist

File: components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb [hunk 12]
```diff
@@ -287,6 +289,19 @@
...
+  "en.faa.incomes.job_income" => 'Job Income',
+  "en.faa.incomes.employer" => 'Employer Name',
+  "en.faa.incomes.employer_phone" => 'Phone Number',
+  "en.faa.incomes.remove_all" => 'Are you sure you want to remove all job income(s)?',
+  "en.faa.incomes.confirm_deletion" => 'Confirm Deletion',
+  "en.faa.incomes.job_income_message" => "Tell us about job income for %{subject}, select “Continue to Next Step” when finished.",
+  "en.faa.incomes.job_end_date_note" => "Note, for %{income_type} income %{subject} currently receives, do not enter an end date into the “End Date” field, only enter an end date if the %{income_type} income has ended.",
+  "en.faa.incomes.from_employer_question" => "Does %{subject} have income from an employer?",
+  "en.faa.incomes.end_date_warning" => "Leave the “%{end_date_field_label}” field blank unless your income will end on a specific date (e.g. the date your seasonal job ends). Only enter the last day of the current year if you won't have this income next year.",
+  "en.faa.incomes.end_date_header" => "End Date:",
```
Comment (charlienparker) [inline components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb:303]: This key is referenced in the new end_date_warning partial, which I'm unsure that we need. If we remove that partial, we can remove this string

File: components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/shared/_end_date_warning.html.erb:1]: This is the pop up modal for 106 reqs

File: components/financial_assistance/app/views/financial_assistance/shared/_date_warning.html.erb [hunk 13]
```diff
@@ -11,7 +11,8 @@
           <%= l10n('faa.start_date_warning', start_date_field_label: l10n('start_date'), current_year: current_year, next_year: next_year) %>
         </li>
         <li id=<%="end_date_warning_#{form_object_id}"%> class="hidden" data-cuke="end-date-warning">
-          <%= l10n('faa.end_date_warning', end_date_field_label: l10n('end_date')) %>
+          <b><%= l10n("faa.incomes.end_date_header") %></b>
+          <%= l10n("faa.incomes.end_date_reminder", end_date_field_label: l10n('end_date')) %>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/shared/_date_warning.html.erb:15]: this key doesn't exist. I think we can keep the `faa.end_date_warning` that was removed on this line, and take out the "End Date" substring from that translation so it can be bolded in the markup as opposed to the translation value.

File: components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb [hunk 14]
```diff
@@ -287,6 +289,19 @@
...
   "en.faa.start_date_warning" => "<strong>Start Date:</strong> This is a required field. If you began receiving this income before this year, enter January 1 of the current year or the income start date. Only enter a future start date if you don't receive this income right now, but expect to receive it in the future.",
   "en.faa.end_date_warning" => "<strong>End Date:</strong> Leave the “%{end_date_field_label}” field blank unless your income will end on a specific date (e.g. the date your seasonal job ends). Only enter the last day of the current year if you won't have this income next year.",
```
Comment (charlienparker) [inline components/financial_assistance/db/seedfiles/translations/en/dc/financial_assistance.rb:291]: This key has been replaced by faa.incomes.end_date_warning. Update any callers of this key to use this new key and the faa.incomes.end_date_header key.

**Review by kristinmerbach — COMMENTED** [review]
Notes from looking in the browser:

- there is extra space between the two top level questions. 
- There are WAVE errors when I open the income form on the buttons: Screenshot 2024-06-28 at 7 38 56 AM. It looks like WAVE doesn't accept that a[:disabled] are exempt from contrast warnings. I think OK for now, but will need to flag
- When you initially land on the page, the inputs are on "yes, with no form" so you have to click no then yes to get the forms to appear
-

**Comment by charlienparker** [comment]
> Notes from looking in the browser:
> 
> * there is extra space between the two top level questions.
> * There are WAVE errors when I open the income form on the buttons: Screenshot 2024-06-28 at 7 38 56 AM. It looks like WAVE doesn't accept that a[:disabled] are exempt from contrast warnings. I think OK for now, but will need to flag
> * When you initially land on the page, the inputs are on "yes, with no form" so you have to click no then yes to get the forms to appear

In regards to the last point, you can replicate that behavior if you check yes to go into the create form experience _and then_ refresh, as well. I think legacy also behaves like that? I looked into fixing this on my PR by checking if there are checked incomes without any created forms on page load and then rendering a create form, but it was a little yucky. Wondering if we open a feature enhancement ticket for that, I'd be open to taking the lead on that.

**Comment by kristinmerbach** [comment]
> Wondering if we open a feature enhancement ticket for that, I'd be open to taking the lead on that.

Yes please.

**Comment by charlienparker** [comment]
> > Wondering if we open a feature enhancement ticket for that, I'd be open to taking the lead on that.
> 
> Yes please.

Done: https://www.pivotaltracker.com/story/show/187876249

**Comment by ATBull81** [comment]
Screenshot 2024-06-28 at 11 16 06 AM
@kristinmerbach @charlienparker spacing should be improved

**Review by kristinmerbach — APPROVED** [review]

File: app/javascript/css/forms.scss [hunk 15]
```diff
@@ -217,6 +217,10 @@ input[type='date']:invalid::-webkit-datetime-edit {
   overflow: hidden;
 }
 
+.income-field {
```
Comment (charlienparker) [inline app/javascript/css/forms.scss:220]: is this needed? we should be able to accomplish width constraints using bootstrap flex

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 16]
```diff
@@ -1,33 +1,40 @@
...
-    // enable "Add New" income buttons
-  $('a.new-income').removeClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').removeAttr('disabled');
+  // enable "Add New" income buttons
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').removeAttr('disabled');
   $("a.interaction-click-control-add-more").removeClass('hide');
   $("a[class*='income-edit'], a[class*='income-delete']").removeClass('disabled'); // legacy
+  $('#self-employed-income').removeClass('disabled');
+  $('a.new-income').removeClass('disabled');
+  $('a.self-employed-income-delete').removeClass('disabled');
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:14]: is this needed? this should be captured by the above a[class*='income-delete'] selector

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 17]
```diff
@@ -1,33 +1,40 @@
...
-  $('a.new-income').addClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').attr('disabled', true);
+  $('#self-employed-income').addClass('disabled');
+  $('a.new-income').addClass('disabled');
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').attr('disabled', true);
   $("a.interaction-click-control-add-more").addClass('hide'); // legacy
 
   // disable all income edit and edit buttons on created incomes
   $("a[class*='income-edit'], a[class*='income-delete']").addClass('disabled');
+  $('a.self-employed-income-delete').addClass('disabled');
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:37]: is this needed? this should be captured by the above a[class*='income-delete'] selector

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb [hunk 18]
```diff
@@ -6,8 +6,8 @@ class IncomesController < FinancialAssistance::ApplicationController
 
     before_action :find_application_and_applicant
     before_action :set_cache_headers, only: [:index, :other]
-    before_action :enable_bs4_layout, only: [:other] if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
-    before_action :conditionally_enable_bs4_layout, only: [:create, :update] if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
+    before_action :enable_bs4_layout, only: [:other, :index, :create] if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
```
Comment (charlienparker) [inline components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:9]: does 'create' need to be declared here if it's already declared in the `conditionally_enable_bs4_layout` before_action?

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb [hunk 19]
```diff
@@ -6,8 +6,8 @@ class IncomesController < FinancialAssistance::ApplicationController
...
+    before_action :conditionally_enable_bs4_layout, only: [:new, :create, :edit, :update] if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
```
Comment (charlienparker) [inline components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:10]: do `new` and `edit` need to be declared here? I don't think they're rendering any UI

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb [hunk 20]
```diff
@@ -139,12 +138,12 @@ def enable_bs4_layout
...
-      when "index", "step", "new"
+      when "step", "new"
         "financial_assistance_nav"
-      when "other"
-        EnrollRegistry.feature_enabled?(:bs4_consumer_flow) ? "financial_assistance_progress" : "financial_assistance"
+      when "index", "other"
+        EnrollRegistry.feature_enabled?(:bs4_consumer_flow) ? "financial_assistance_progress" : "financial_assistance_nav"
       else
-        "financial_assistance"
+        "financial_assistance_nav"
```
Comment (charlienparker) [inline components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:146]: is this correct? which actions should be using this layout irrespective of the flag?

seems weird to change this when they were using `financial_assistance` before

Comment (ATBull81) [reply components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:146]: without this change, there is no nav for non-bootstrap layout

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 21]
```diff
@@ -1,33 +1,40 @@
...
   $('.driver-question, .driver-question a, .instruction-row, .income:not(#' + income_kind + '), .other-income-kind:not(#' + income_kind + '), .unemployment-income:not(#' + income_kind + ')').addClass('disabled');
   // disable inputs on drivers, instructions, and other income kinds not being edited - except for submit (save) buttons as some form types require them to be enabled on edit/create
   $('.driver-question input, .instruction-row input, .income:not(#' + income_kind + ') input:not(":input[type=submit]"), .other-income-kind:not(#' + income_kind + ') input:not(":input[type=submit]"), .unemployment-income:not(#' + income_kind + ') input:not(":input[type=submit]")').attr('disabled', true);
-  
+    
   // disable "Add New" income buttons
-  $('a.new-income').addClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').attr('disabled', true);
+  $('#self-employed-income').addClass('disabled');
+  $('a.new-income').addClass('disabled');
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:31]: is this not captured by the `.new-income` selector on line 32?

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 22]
```diff
@@ -1,33 +1,40 @@
...
   // apply disabled style to drivers, instructions, and other income kinds not being edited
   $('.driver-question, .driver-question a, .instruction-row, .income:not(#' + income_kind + '), .other-income-kind:not(#' + income_kind + '), .unemployment-income:not(#' + income_kind + ')').addClass('disabled');
   // disable inputs on drivers, instructions, and other income kinds not being edited - except for submit (save) buttons as some form types require them to be enabled on edit/create
   $('.driver-question input, .instruction-row input, .income:not(#' + income_kind + ') input:not(":input[type=submit]"), .other-income-kind:not(#' + income_kind + ') input:not(":input[type=submit]"), .unemployment-income:not(#' + income_kind + ') input:not(":input[type=submit]")').attr('disabled', true);
-  
+    
   // disable "Add New" income buttons
-  $('a.new-income').addClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').attr('disabled', true);
+  $('#self-employed-income').addClass('disabled');
```
Comment (charlienparker) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:30]: I think for consistency this should be moved to the long selector on line 25

File: app/javascript/css/forms.scss (hunk 15 above)
Comment (ATBull81) [reply app/javascript/css/forms.scss:220]: Bootstrap flex options do not split the row 50/50 no matter which flex class is used, easiest solution was to introduce a new class

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 21 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:31]: This is not captured; without it the button is still clickable and can add another new income/self-employed form

Comment (charlienparker) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:31]: Interesting, I removed it on my local without seeing that issue

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb [hunk 23]
```diff
@@ -1,77 +1,125 @@
...
+    <div class="display-income">
+      <div class="focus_effect_module">
+        <div class="d-flex col-sm income-form-row p-0 mb-2">
+          <div class="w-100 pr-3">
+            <label class="static_label label-floatlabel required" for="name-<%= income.employer_name %>"><%= l10n("faa.incomes.employer") %></label>
+            <input class="fake-disabled-input w-100" id="name-<%= income.employer_name %>" disabled="true" value="<%= income.employer_name %>">
+          </div>
+          <div class="w-100">
+            <label class="static_label label-floatlabel required" for="phone-<%= income.employer_phone %>-<%= income.id %>"><%= l10n("faa.incomes.employer_phone") %></label>
+            <input class="fake-disabled-input w-100" id="phone-<%= income.employer_phone %>-<%= income.id %>" disabled="true" value="<%= income.employer_phone %>">
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb:12]: should probably add a mask on this phone input, I know QA has been pushing for that

File: app/javascript/css/forms.scss (hunk 15 above)
Comment (charlienparker) [reply app/javascript/css/forms.scss:229]: I think d-flex and cols can be used exactly for this purpose. I just tried on my local and it works. May need some extra work to get smaller breakpoints to play nice, but not a huge deal there probably

File: components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb [hunk 24]
```diff
@@ -1,212 +1,441 @@
...
+          <%= render partial: 'financial_assistance/incomes/self_employed_income', locals: { income: income }%>
         <% end %>
       </div>
-
-      <a tabindex="0" onkeydown="handleButtonKeyDown(event, 'new-self-income')" id='new-self-income' class="new-income"> <span class="fa-icon fa-stack plus-mr" aria-hidden='true'><i class="fa fa-circle-thin fa-stack-2x"></i><i class="fa fa-plus fa-stack-1x"></i></span></span> <b>Add Another </b> Self-Employed Income </a>
-
-      <div id="hidden-income-form" class='new-income-form hidden'>
-        <%= render partial: 'financial_assistance/incomes/self_employed_income_form', locals: { income: @applicant.incomes.build, new_form: true} %>
+      <div class="add-more-link mt-3">
+        <a tabindex="0" onkeydown="handleButtonKeyDown(event, 'self-employed-income', 'new-income-form')" id="self-employed-income" class='<%= 'hidden' unless @applicant.incomes.self_employment.present? %> new-income new-income-self-employed button mb-4'><%= l10n("faa.other_incomes.add", kind: "Self-Employed") %></a>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb:73]: "Self-Employed" should be localized

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb [hunk 25]
```diff
@@ -1,78 +1,149 @@
...
+      <div class="mt-1 pl-0 pr-2 pt-1">
+        <p class="subtext"><%= l10n('faa.note') %> <%= l10n('faa.incomes.income_note', subject: @applicant.first_name, income_kind: "job income", end_date_field_label: l10n('end_date')) %></p>
       </div>
     </div>
-    <div class="row row-form-wrapper no-buffer row-height">
-      <div class="col-md-3">
-        <%= f.label :frequency_kind, 'Amount *', class: "amount label-color" %>
-        <div class="faa-amount">
-          <%= f.text_field :amount, placeholder: '0.00', value: number_to_currency(income.amount), required: true %>
+    <div class="focus_effect_module row-edit-border pl-3 pr-3 mb-1">
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb:17]: legacy classes should be removed

File: components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income.html.erb [hunk 26]
```diff
@@ -1,41 +1,78 @@
...
+            <label class="static_label label-floatlabel required" for="name-<%= income.amount %>" ><%= l10n("faa.other_incomes.amount") %></label>
+            <input class="fake-disabled-input w-100" id="name-<%= income.amount %>" disabled="true" value="<%= number_to_currency(income.amount) %>">
+          </div>
 
-        <div class="col-md-3 no-pd form-group-lg no-pd class-fa-household">
-          <label class="static_label label-floatlabel">HOW OFTEN</label>
-          <span class="field_value floatlabel form-control active-floatlabel" id="name"><%= frequency_kind_options.invert[income.frequency_kind] %></span>
-        </div>
+          <div class="w-100 pr-3">
+            <label class="static_label label-floatlabel required" for="name-<%= frequency_kind_options.invert[income.frequency_kind] %>-<%=income.id%>"><%= l10n("faa.other_incomes.frequency") %></label>
```
Comment (charlienparker) [inline components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income.html.erb:12]: legacy classes should be removed

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb (hunk 19 above)
Comment (kristinmerbach) [reply components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:10]: New and edit render js that switched out the form vs the show for the income elements

Comment (charlienparker) [reply components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb:10]: @kristinmerbach on my local, it doesn't look like these endpoints are even called. I think they were switched out at some point for some in-place JS

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 22 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js]: Seems like it's getting covered by another item in that line, was able to remove line 30 and fields are behaving as intended

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 21 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js]: Wasn't working for me but I removed it; we will need to fix if it is broken on bs4 after merge

File: components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income.html.erb (hunk 26 above)
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/incomes/_self_employed_income.html.erb]: removed static_label and label-floatlabel throughout

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb (hunk 25 above)
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/incomes/_job_income_form.html.erb]: removed focus_effect_module

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb [hunk 27]
```diff
@@ -1,77 +1,125 @@
...
-          <a tabindex="0" onkeydown="handleButtonKeyDown(event, 'income-delete-<%= dom_id income %>')" id="income-delete-<%= dom_id income %>" class="income-delete"><i class="fa fa-trash-o fa-lg" aria-hidden="true"></i>
-            <span class="hide"><%=l10n("delete")%></span>
-          </a>
+<% if @bs4 %>
+  <div id="<%= dom_id income %>" class="income my-4">
+    <div class="display-income">
+      <div class="focus_effect_module">
+        <div class="d-flex col-sm income-form-row p-0 mb-2">
+          <div class="w-100 pr-3">
+            <label class="static_label label-floatlabel required" for="name-<%= income.employer_name %>"><%= l10n("faa.incomes.employer") %></label>
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb]: removed static_label label-floatlabel throughout

File: components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb (hunk 24 above)
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/incomes/index.html.erb]: fixed

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb (hunk 19 above)
Comment (ATBull81) [reply components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb]: removed both as it was working fine on local without either

File: app/javascript/css/forms.scss (hunk 15 above)
Comment (ATBull81) [reply app/javascript/css/forms.scss]: removed after refactor

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 16 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js]: removed

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 17 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js]: removed

File: components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb (hunk 18 above)
Comment (ATBull81) [reply components/financial_assistance/app/controllers/financial_assistance/incomes_controller.rb]: updated and removed

**Comment by ATBull81** [comment]
Screenshot 2024-06-28 at 4 50 07 PM

**Comment by ATBull81** [comment]
Screenshot 2024-06-28 at 4 58 23 PM

File: components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb (hunk 23 above)
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/incomes/_job_income.html.erb]: masking added for (000) 000-0000 format

**Review by charlienparker — APPROVED** [review]

**Review by kristinmerbach — APPROVED** [review]
